### PR TITLE
feat: Support to add random seed on random engine

### LINKED ISSUE #11863: Feature:  Support to add random seed on random engine
**Summary**
Create table(t int, time timestamp) Engine=Random(1000)

The sql would create a generate random table with random seed 1000, such feature would deterministically test a lot of cases need randome data

### PR DESCRIPTION
**Pull request by blackstar-baba**
I hereby agree to the terms of the CLA available at: https://docs.databend.com/dev/policies/cla/

## Summary

Syntax:
```
create table t(a int, time timestamp) Engine=Random(1000);
select * from t; -- -- this will generate 1000 random values.
```

- Fixes #11863

## Tests

- [x] Unit Test
- [ ] Logic Test
- [ ] Benchmark Test
- [ ] No Test  - _Explain why_

## Type of change

- [ ] Bug Fix (non-breaking change which fixes an issue)
- [x] New Feature (non-breaking change which adds functionality)
- [ ] Breaking Change (fix or feature that could cause existing functionality not to work as expected)
- [ ] Documentation Update
- [ ] Refactoring
- [ ] Performance Improvement
- [ ] Other (please describe):

- - -
This change is [Reviewable](https://reviewable.io/reviews/datafuselabs/databend/15167)

### CODE REVIEW

**Comment by blackstar-baba** [comment]
I will add some test cases.

**Comment by PsiACE** [comment]
Cool, after this PR, more reproducible examples can be added to sqllogictest.

File: src/query/expression/src/values.rs [hunk 1]
```diff
@@ -1311,6 +1310,159 @@ impl Column {
         }
     }
 
+    pub fn random_with_seed(ty: &DataType, len: usize, seed: u64) -> Self {
```
Comment (sundy-li) [inline src/query/expression/src/values.rs:1313]: maybe we can replace the old `random` function with a fixed seed param?

**Review by sundy-li — APPROVED** [review]

**Comment by BohuTANG** [comment]
@blackstar-baba Good job! 
It's would be greate to add how to use it in the PR summary.

**Comment by blackstar-baba** [comment]
> > @blackstar-baba Good job! It's would be great to add how to use it in the PR summary.
> 
> Updated the PR summary.

Modified.

**Pull request merged**